In [1]:
import pandas as pd

In [6]:
df = pd.read_pickle("../../data/dummy.pkl")
df

state  account length international plan voice mail plan  \
0       KS             128                 no             yes   
1       OH             107                 no             yes   
2       NJ             137                 no              no   
3       OH              84                yes              no   
4       OK              75                yes              no   
...    ...             ...                ...             ...   
3328    AZ             192                 no             yes   
3329    WV              68                 no              no   
3330    RI              28                 no              no   
3331    CT             184                yes              no   
3332    TN              74                 no             yes   

      number vmail messages  total day minutes  total day calls  \
0                        25              265.1              110   
1                        26              161.6              123   
2                         0              243.4              114   
3                         0              299.4               71   
4                         0              166.7              113   
...                     ...                ...              ...   
3328                     36              156.2               77   
3329                      0              231.1               57   
3330                      0              180.8              109   
3331                      0              213.8              105   
3332                     25              234.4              113   

      total day charge  total eve minutes  total eve calls  ...  \
0                45.07              197.4               99  ...   
1                27.47              195.5              103  ...   
2                41.38              121.2              110  ...   
3                50.90               61.9               88  ...   
4                28.34              148.3              122  ...   
...                ...                ...              ...  ...   
3328             26.55              215.5              126  ...   
3329             39.29              153.4               55  ...   
3330             30.74              288.8               58  ...   
3331             36.35              159.6               84  ...   
3332             39.85              265.9               82  ...   

      total night charge  total intl minutes  total intl calls  \
0                  11.01                10.0                 3   
1                  11.45                13.7                 3   
2                   7.32                12.2                 5   
3                   8.86                 6.6                 7   
4                   8.41                10.1                 3   
...                  ...                 ...               ...   
3328               12.56                 9.9                 6   
3329                8.61                 9.6                 4   
3330                8.64                14.1                 6   
3331                6.26                 5.0                10   
3332               10.86                13.7                 4   

      total intl charge  customer service calls  churn  \
0                  2.70                       1      0   
1                  3.70                       1      0   
2                  3.29                       0      0   
3                  1.78                       2      0   
4                  2.73                       3      0   
...                 ...                     ...    ...   
3328               2.67                       2      0   
3329               2.59                       3      0   
3330               3.81                       2      0   
3331               1.35                       2      0   
3332               3.70                       0      0   

      total_domestic_minutes  total_domestic_calls  total_domestic_charge  \
0                      707.2                   

In [7]:
df["churn"].value_counts(normalize=True)

0    0.855086
1    0.144914
Name: churn, dtype: float64

The model would only be right 14.5% of the time if the model only guessed every customer would leave.

As we devlop our models, we will be optomizing the recall score of our models. Our strategies We may identify more people as leaving, but in reality they stay. The company will spend some resources to retain the customer. However, that the cost of keeping them engaged with the company far outweighs the cost of identifying someone as staying and in actuality they leave the company. The company would lose a customer and then have to spend more resources to get a new customer in their place.